### PR TITLE
Fix inconsistency on date fields and add index on name fields

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

1. Add a checks on date fields consistency (`date_creation` and `date_mod` should not be used separately).
2. Fields that store a valuable date should be named `date`.
3. Add a check to ensure `name` fields are indexed (as this is the default field used for sorting lists).

### CODE REVIEW

File: inc/knowbaseitem.class.php [hunk 1]
```diff
@@ -1859,15 +1853,6 @@ function rawSearchOptions() {
          'datatype'           => 'dropdown'
       ];
 
-      $tab[] = [
```
Comment (trasher) [inline inc/knowbaseitem.class.php:1862]: You have to take care of use preferences table, and saved searches

Comment (cedric-anne) [reply inc/knowbaseitem.class.php:1862]: Done in last commit.

**Review by orthagh — COMMENTED** [review]
i checked at least kb, i have a sql error on index

```
[2021-06-24 10:48:26] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/10.0.git/inc/dbmysql.class.php line 337
  *** MySQL query error:
  SQL: SELECT DISTINCT `glpi_knowbaseitems`.* FROM `glpi_knowbaseitems` LEFT JOIN `glpi_knowbaseitems_users` ON (`glpi_knowbaseitems_users`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_groups_knowbaseitems` ON (`glpi_groups_knowbaseitems`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_knowbaseitems_profiles` ON (`glpi_knowbaseitems_profiles`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_entities_knowbaseitems` ON (`glpi_entities_knowbaseitems`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) WHERE ( NOT (`glpi_entities_knowbaseitems`.`entities_id` IS NULL AND `glpi_knowbaseitems_profiles`.`profiles_id` IS NULL AND `glpi_groups_knowbaseitems`.`groups_id` IS NULL AND `glpi_knowbaseitems_users`.`users_id` IS NULL)) AND (((`glpi_knowbaseitems`.`begin_date` IS NULL) OR (`glpi_knowbaseitems`.`begin_date` < NOW()))) AND (((`glpi_knowbaseitems`.`end_date` IS NULL) OR (`glpi_knowbaseitems`.`end_date` > NOW()))) ORDER BY `date` DESC LIMIT 10
  Error: Unknown column 'date' in 'order clause'
  Backtrace :
  inc/dbmysqliterator.class.php:95                   
  inc/dbmysql.class.php:821                          DBmysqlIterator->execute()
  inc/knowbaseitem.class.php:1805                    DBmysql->request()
  inc/knowbase.class.php:127                         KnowbaseItem::showRecentPopular()
  inc/knowbase.class.php:82                          Knowbase::showSearchView()
  inc/commonglpi.class.php:656                       Knowbase::displayTabContentForItem()
  ajax/common.tabs.php:106                           CommonGLPI::displayStandardTab()
  {"user":"21@DESKTOP-CUCL3BV"} 
```

**Comment by cedric-anne** [comment]
> i checked at least kb, i have a sql error on index
> 
Fixed in latest commit.

File: inc/notificationtargetobjectlock.class.php [hunk 2]
```diff
@@ -110,8 +111,9 @@ function addDataForTemplate($event, $options = []) {
       $this->data['##objectlock.name##']     = $object->fields['name'];
       $this->data['##objectlock.id##']       = $options['item']->fields['items_id'];
       $this->data['##objectlock.type##']     = $options['item']->fields['itemtype'];
-      $this->data['##objectlock.date_mod##'] = Html::convDateTime($options['item']->fields['date_mod'],
+      $this->data['##objectlock.date##']     = Html::convDateTime($options['item']->fields['date_mod'],
                                                                    $user->fields['date_format']);
+      $this->data['##objectlock.date_mod##'] = $this->data['##objectlock.date##'];
```
Comment (trasher) [inline inc/notificationtargetobjectlock.class.php:116]: Should not you use `Html::conDateTime`?
Also, `##objectlock.date##` on `date_mod` and `##objectlock.date_mod##` on date sounds a bit weird

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**